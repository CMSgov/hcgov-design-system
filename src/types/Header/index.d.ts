import React from "react";
import { Language } from "../index";

interface Link {
  href: string,
  label: React.ReactNode,
  onClick?: (...args: any[]) => any,
}

export interface HeaderProps {
  className?: string;
  inverse?: boolean;
  inversed?: boolean;
  initialLanguage?: Language;
  switchLocaleLink?: string;
  loggedIn?: boolean;
  firstName?: React.ReactNode;
  subhead?: React.ReactNode;
  subpath?: string;
  primaryDomain?: string;
  skipNavHref?: string;
  onSkipNavClick?: func;
  deConsumer?: boolean;
  deBrokerName?: string;
  links?: Link[];
}

export class Header extends React.Component<HeaderProps, any> {
  render(): JSX.Element;
}

export default Header;

interface DefaultLinks {
  home: Link[];
  minimal: Link[];
  'logged-in': Link[];
}

export function defaultMenuLinks(
  locale: Language,
  deConsumer: boolean,
  subpath: string,
  primaryDomain: string = '',
  switchLocaleLink?: string
): DefaultLinks;

