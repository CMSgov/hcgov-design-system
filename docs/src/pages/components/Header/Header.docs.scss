/*
Header

The HealthCare.gov header component varies depending on the user's state (ie. logged-in?) and context (ie. what page is this user on?).

Style guide: patterns.header
*/
/*
`<Header />`

@react-example Header.example.jsx

@react-props Header.jsx

@responsive

Style guide: patterns.header.react
*/
