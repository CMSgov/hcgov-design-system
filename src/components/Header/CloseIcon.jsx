import React from 'react';
import closeSvg from '@cmsgov/design-system/dist/images/close--primary.svg';

/**
 * wrapper around the core close icon. Not sure I'm happy with this implementation
 * @param {string} className providide additional class strings for the close icon
 * @returns {JSX}
 */

export function CloseIcon({ className }) {
  return (
    <span
      dangerouslySetInnerHTML={{ __html: closeSvg }}
      className={`hc-c-header__close-icon ds-u-valign--middle ${className}`}
    />
  );
}

export default CloseIcon;
