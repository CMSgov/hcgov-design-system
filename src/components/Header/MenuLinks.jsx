import PropTypes from 'prop-types';
import React from 'react';

/**
 * The primary list of menu links. This ensures a consistent styling
 * across header variations. On the homepage, this list is displayed
 * horizontally on desktop. In other versions, this list is visible
 * when the "Menu" button is toggled and displayed as a stacked list.
 */
const MenuLinks = function (props) {
  return (
    <ul className="hc-c-menu__links ds-c-list ds-c-list--bare">
      {props.links.map(function (link) {
        return (
          <li key={link.href} className="ds-u-margin--0">
            <a
              href={link.href}
              onClick={link.onClick}
              className="hc-c-menu__link"
            >
              {link.label}
            </a>
          </li>
        );
      })}
    </ul>
  );
};

MenuLinks.propTypes = {
  links: PropTypes.arrayOf(
    PropTypes.shape({
      href: PropTypes.string.isRequired,
      label: PropTypes.node.isRequired,
      onClick: PropTypes.func,
    })
  ).isRequired,
};

export default MenuLinks;
