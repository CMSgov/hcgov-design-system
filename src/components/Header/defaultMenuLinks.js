import localeLink from './localeLink';
import loginLink from './loginLink';
import { translate } from '../i18n';

/**
 * Default menu links for each header variation.
 * Apps can import this method into their app if they need to
 * extend the existing default list of menu links.
 * @param {Function} translate - i18n translator
 * @param {String} locale
 * @param {Boolean} deConsumer
 * @param {String} subpath
 * @param {String} primaryDomain
 * @param {String} switchLocaleLink
 * @param {boolean} hideLoginLink
 * @param {boolean} hideLanguageSwitch
 * @returns {Object}
 */
export function defaultMenuLinks(
  locale = 'en',
  deConsumer,
  subpath,
  primaryDomain = '',
  switchLocaleLink,
  hideLoginLink,
  hideLanguageSwitch
) {
  const isSpanish = locale === 'es';
  const ffmLocalePath = isSpanish ? 'es_MX' : 'en_US';
  // We import and set i18n options within this method, for
  // scenarios where an app needs to import this method and
  // extend the existing list of default links
  const t = translate;
  const i18nOptions = { lng: locale };

  const minimal = [];
  if (!hideLanguageSwitch) minimal.push(localeLink(t, locale, subpath, switchLocaleLink));
  if (!hideLoginLink) minimal.push(loginLink(t, deConsumer, primaryDomain));

  return {
    'logged-out': minimal,
    'logged-in': [
      {
        label: t('header.myApplicationsAndCoverage', i18nOptions),
        href: `${primaryDomain}/marketplace/auth/global/${ffmLocalePath}/myProfile#landingPage`,
      },
      {
        label: t('header.myProfile', i18nOptions),
        href: `${primaryDomain}/marketplace/auth/global/${ffmLocalePath}/myProfile#settings`,
      },
    ],
  };
}

export default defaultMenuLinks;
