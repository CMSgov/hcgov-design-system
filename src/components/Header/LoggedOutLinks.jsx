import PropTypes from 'prop-types';
import React from 'react';
import defaultMenuLinks from './defaultMenuLinks';

/**
 * The list of links displayed horizontally in the header in the logged-out
 * menu. This by default is the login link and the language link.
 */
const LoggedOutLinks = ({
  locale,
  deConsumer,
  subpath,
  primaryDomain,
  switchLocaleLink,
}) => {
  const links = defaultMenuLinks(
    locale,
    deConsumer,
    subpath,
    primaryDomain,
    switchLocaleLink
  ).minimal;
  return (
    <ul className="hc-c-logged-out-links ds-c-list--bare ds-u-display--none ds-u-sm-display--inline-block">
      {links.map((link) => {
        return (
          <li key={link.href} className="hc-c-logged-out-links__li">
            <a href={link.href} className="hc-c-logged-out-links__link">
              {link.label}
            </a>
          </li>
        );
      })}
    </ul>
  );
};

LoggedOutLinks.propTypes = {
  locale: PropTypes.string.isRequired,
  /**
   * Indicates when a consumer is coming from a Direct Enrollment flow.
   * This will include additional messaging and modify some of the links.
   */
  deConsumer: PropTypes.bool,
  /**
   * [See `Header.props.subpath`]({{root}}/patterns/header/#patterns.header.react)
   */
  subpath: PropTypes.string,
};

export default LoggedOutLinks;
