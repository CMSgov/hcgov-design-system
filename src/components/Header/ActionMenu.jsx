/* eslint-disable react/no-multi-comp, jsx-a11y/no-redundant-roles */
import { Button } from '@cmsgov/design-system';
import LoggedOutLinks from './LoggedOutLinks';
import PropTypes from 'prop-types';
import React from 'react';
import classnames from 'classnames';
import { translate } from 'react-i18next';

const menuId = 'hc-c-menu';

/**
 * ActionMenu is displayed at the top-right of the header.
 * There are two variations of this component: One for logged-out
 * users, which displays a "Log in" & "Espanol" link on desktop.
 * And another for logged-in users, which includes the user's first
 * name. The logged-in variation always displays a "Menu" toggle
 * button, and the logged-out variation only displays it on mobile.
 */
const ActionMenu = function (props) {
  return (
    <nav
      id="hc-c-header__actions"
      className="hc-c-header__actions ds-l-col ds-l-col--auto ds-u-margin-left--auto ds-u-font-weight--bold"
      role="navigation"
    >
      {props.loggedIn ? (
        <LoggedInActionMenu {...props} />
      ) : (
        <LoggedOutActionMenu {...props} />
      )}
    </nav>
  );
};

function LoggedInActionMenu(props) {
  return (
    <div>
      {props.firstName && (
        <span className="ds-u-display--none ds-u-sm-display--inline-block">
          {props.firstName}
        </span>
      )}
      <MenuButton {...props} />
    </div>
  );
}

function LoggedOutActionMenu(props) {
  return (
    <div>
      <LoggedOutLinks
        locale={props.locale}
        deConsumer={props.deConsumer}
        subpath={props.subpath}
        primaryDomain={props.primaryDomain}
        switchLocaleLink={props.switchLocaleLink}
      />
      <MenuButton
        {...props}
        className="ds-u-display--inline-block ds-u-sm-display--none"
      />
    </div>
  );
}

function MenuButton({ t, open, ...props }) {
  const className = classnames(
    'hc-c-action-menu-button',
    'ds-u-font-weight--bold',
    'ds-u-sm-margin-left--2',
    props.className
  );
  return (
    <Button
      inversed={props.inversed}
      aria-controls={menuId}
      aria-expanded={!!open}
      aria-label={open ? t('header.closeMenu') : t('header.openMenu')}
      className={className}
      onClick={props.onMenuToggleClick}
      size="small"
    >
      {open ? t('header.close') : t('header.menu')}
    </Button>
  );
}

ActionMenu.propTypes = {
  /** Applies the inverse theme styling */
  inversed: PropTypes.bool,
  firstName: PropTypes.string,
  locale: PropTypes.string.isRequired,
  loggedIn: PropTypes.bool,
  onMenuToggleClick: PropTypes.func.isRequired,
  /**
   * Indicates the menu is open, which influences the label
   * and ARIA attribute of the toggle button.
   */
  open: PropTypes.bool,
  /**
   * Indicates when a consumer is coming from a Direct Enrollment flow.
   * This will include additional messaging and modify some of the links.
   */
  deConsumer: PropTypes.bool,
  /**
   * [See `Header.props.subpath`]({{root}}/patterns/header/#patterns.header.react)
   */
  subpath: PropTypes.string,
};

LoggedInActionMenu.propTypes = {
  firstName: ActionMenu.propTypes.firstName,
  onMenuToggleClick: ActionMenu.propTypes.onMenuToggleClick,
  open: ActionMenu.propTypes.open,
};

LoggedOutActionMenu.propTypes = {
  onMenuToggleClick: ActionMenu.propTypes.onMenuToggleClick,
  open: ActionMenu.propTypes.open,
  deConsumer: ActionMenu.propTypes.deConsumer,
  locale: ActionMenu.propTypes.locale,
};

MenuButton.propTypes = {
  onMenuToggleClick: ActionMenu.propTypes.onMenuToggleClick,
  open: ActionMenu.propTypes.open,
  className: PropTypes.string,
};

export default translate()(ActionMenu);
