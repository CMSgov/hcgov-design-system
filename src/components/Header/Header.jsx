import ActionMenu from './ActionMenu';
import DeConsumerMessage from './DeConsumerMessage';
import Logo from '../Logo/Logo';
import Menu from './Menu';
import PropTypes from 'prop-types';
import React from 'react';
import { SkipNav } from '@cmsgov/design-system';
import classnames from 'classnames';
import defaultMenuLinks from './defaultMenuLinks';
import localeLink from './localeLink';
import { translate } from 'react-i18next';
import { translate as translateWrapper } from '../i18n';

export const LOGGED_IN_VAL = 'logged-in';

/**
 * The top-level component, responsible for maintaining the
 * header's state (like whether the mobile menu is expanded) and
 * determining which variation of the header to display
 */
export class _Header extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      openMenu: false,
      hideLoginLink: false,
      hideLanguageSwitch: false,
    };
  }

  isLoggedIn() {
    return this.variation() === LOGGED_IN_VAL;
  }

  /**
   * Content rendered within <Menu>, after the list of links
   * @returns {Node}
   */
  afterMenuLinks() {
    if (this.isLoggedIn()) {
      const { initialLanguage = 'en', primaryDomain } = this.props;
      const i18nOptions = { lng: initialLanguage };

      return (
        <a
          className="hc-c-menu__link ds-u-border-top--1 ds-u-margin-x--1 ds-u-padding-x--0"
          href={`${primaryDomain}/logout`}
        >
          {translateWrapper('header.logout', i18nOptions)}
        </a>
      );
    }
  }

  /**
   * Content rendered within <Menu>, before the list of links
   * @returns {Node}
   */
  beforeMenuLinks() {
    if (this.isLoggedIn() && this.props.firstName) {
      return (
        <div className="ds-u-sm-display--none ds-u-border-bottom--1 ds-u-margin-x--1 ds-u-padding-y--1 hc-c-header__name">
          {this.props.firstName}
        </div>
      );
    }
  }

  /**
   * Event handler for when the "Menu" or "Close" button
   * within ActionMenu is clicked.
   */
  handleMenuToggleClick = () => {
    this.setState({ openMenu: !this.state.openMenu });
  };

  /**
   * Determines which variation of the header should be displayed,
   * based on the props being passed into the component.
   * @returns {String} Variation name
   */
  variation() {
    if (this.props.loggedIn) {
      // Logged-in state, with minimal navigation
      return LOGGED_IN_VAL;
    } else if (this.props.subhead) {
      // Product, with minimal navigation and a subheading
      return 'minimal';
    }

    return 'home';
  }

  // if header is in authenticated state, add language link to the end of other links
  // if unauthenticated, return original link list
  getLinksWithLocale(links) {
    if (this.isLoggedIn()) {
      const localeLinkItem = localeLink(
        this.props.t,
        this.props.initialLanguage,
        this.props.subpath,
        this.props.switchLocaleLink
      );

      return [...links, localeLinkItem];
    }
    return links;
  }

  render() {
    const classes = classnames(
      `hc-c-header hc-c-header--${this.variation()}`,
      this.props.className
    );

    const links =
      this.props.links ||
      defaultMenuLinks(
        this.props.initialLanguage,
        this.props.deConsumer,
        this.props.subpath,
        this.props.primaryDomain,
        this.props.switchLocaleLink,
        this.props.hideLoginLink,
        this.props.hideLanguageSwitch
      )[this.variation()];
    const linksWithLocale = this.getLinksWithLocale(links);

    return (
      <header className={classes} role="banner">
        <SkipNav href={this.props.skipNavHref} onClick={this.props.onSkipNavClick}>
          {this.props.t('header.skipNav')}
        </SkipNav>

        <div className="ds-l-container">
          <div className="ds-l-row ds-u-align-items--center ds-u-flex-wrap--nowrap ds-u-padding-y--2">
            <a
              href={this.props.primaryDomain ? this.props.primaryDomain : '/'}
              className="hc-c-logo-link ds-l-col ds-l-col--auto"
            >
              <Logo locale={this.props.initialLanguage} />
            </a>

            {this.props.subhead && (
              <div className="hc-c-header__subhead hc-c-header__subhead--inline ds-u-md-display--block ds-u-display--none">
                {this.props.subhead}
              </div>
            )}

            <ActionMenu
              firstName={this.props.firstName}
              onMenuToggleClick={this.handleMenuToggleClick}
              locale={this.props.initialLanguage}
              loggedIn={this.props.loggedIn}
              hideLoginLink={this.props.hideLoginLink}
              hideLanguageSwitch={this.props.hideLanguageSwitch}
              open={this.state.openMenu}
              deConsumer={this.props.deConsumer}
              subpath={this.props.subpath}
              primaryDomain={this.props.primaryDomain}
              switchLocaleLink={this.props.switchLocaleLink}
            />
          </div>
        </div>

        {this.props.subhead && (
          <div className="hc-c-header__subhead ds-l-container ds-u-display--block ds-u-md-display--none ds-u-padding-y--1 ds-u-border-top--1 ds-u-border-color--inverse">
            {this.props.subhead}
          </div>
        )}

        <Menu
          afterLinks={this.afterMenuLinks()}
          beforeLinks={this.beforeMenuLinks()}
          links={linksWithLocale}
          open={this.state.openMenu}
          primaryDomain={this.props.primaryDomain}
        />

        {this.props.deConsumer && <DeConsumerMessage deBrokerName={this.props.deBrokerName} />}
      </header>
    );
  }
}

_Header.defaultProps = {
  initialLanguage: 'en',
  skipNavHref: '#main',
};

/* eslint-disable react/no-unused-prop-types */
_Header.propTypes = {
  /**
   * Additional classes to be added to the root `<header>` element.
   */
  className: PropTypes.string,
  /**
   * The language the header will render as.
   */
  initialLanguage: PropTypes.oneOf(['en', 'es']),
  /**
   * For applications that handle their own locale switching. Overrides the
   * default locale link. The link's label is still determined by the opposite
   * of the `initialLanguage` provided, i.e. if `initialLanguage` is `en`,
   * the link's label will always be "Espa√±ol". This takes precedence over the
   * `subpath` prop.
   */
  switchLocaleLink: PropTypes.string,
  /**
   * Indicate that a user is logged-in.
   */
  loggedIn: PropTypes.bool,
  /**
   * When set to true, do not display the Login text in the upper right of the
   * header
   */
  hideLoginLink: PropTypes.bool,
  /**
   * When set to true, do not display the the switch locale link
   */
  hideLanguageSwitch: PropTypes.bool,
  /**
   * For logged-in users, pass in their first name to display in the header
   */
  firstName: PropTypes.node,
  /**
   * For product headers, pass in the product name to display next to the logo.
   */
  subhead: PropTypes.node,
  /**
   * For applications hosted at paths other than the root `healthcare.gov`/
   * `cuidadodesalud.gov`. This string will be appended to the end of the
   * language links so as to keep the user within the same part of the site
   * when switching languages.
   */
  subpath: PropTypes.string,
  /**
   * The primary, or root domain where the majority of header links should be
   * hosted.  By default, links render with relative paths, but providing this
   * prop will force all links to render with absolute paths based on the
   * provided string. The string should include the protocol (`http://` or
   * `https://`) and the domain only, with no trailing slash. For example, if
   * the provided string is `https://test.healthcare.gov`, the login link will
   * render as `https://test.healthcare.gov/login` instead of just `/login`.
   * Note that this is only really necessary if your application is hosted on a
   * subdomain, such as `https://localhelp.healthcare.gov`, where relative links
   * would direct the user to the wrong location, e.g. the link to `/login`
   * would incorrectly direct the user to
   * `https://localhelp.healthcare.gov/login` when it should direct the user to
   * `https://healthcare.gov/login`.
   */
  primaryDomain: PropTypes.string,
  /**
   * A URL hash used for the "Skip to main content" link. This is
   * typically the `id` of your "main" content area (ie. `#main`).
   */
  skipNavHref: PropTypes.string,
  /**
   * An onClick handler used for the "Skip to main content" link. This can
   * be used in cases where one would need to manually set the focus on the
   * content area (e.g. where hash routing is being used).
   */
  onSkipNavClick: PropTypes.func,
  /**
   * Indicates when a consumer is coming from a Direct Enrollment flow.
   * This will include additional messaging and modify some of the links.
   */
  deConsumer: PropTypes.bool,
  /**
   * Used in conjunction with `deConsumer`, the Direct Enrollment broker's
   * name is used in some of the messaging displayed to the consumer.
   */
  deBrokerName: PropTypes.string,
  /**
   * Optionally pass in an array of link objects to override the default
   * set of menu links. This may be useful if you need to customize the
   * links on a page-by-page basis. To reference the default set of menu
   * links, you can import the `defaultMenuLinks` method.
   */
  links: PropTypes.arrayOf(
    PropTypes.shape({
      href: PropTypes.string.isRequired,
      label: PropTypes.node.isRequired,
      onClick: PropTypes.func,
    })
  ),
};

export const Header = translate()(_Header);
export default Header;
