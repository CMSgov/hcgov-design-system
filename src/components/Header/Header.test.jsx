import { _Header as Header } from './Header';
import React from 'react';
import { SkipNav } from '@cmsgov/design-system';
import { shallow } from 'enzyme';

jest.mock('../i18n', () => ({
  translate: (text) => text,
}));

function render(props) {
  props = Object.assign(
    {
      t: (key) => key,
    },
    props
  );

  return shallow(<Header {...props} />);
}

describe('Header', function () {
  it('renders full/homepage header', () => {
    expect(render()).toMatchSnapshot();
  });

  it('renders minimal/product header', () => {
    expect(render({ subhead: 'Tax tools' })).toMatchSnapshot();
  });

  it('renders Direct Enrollment banner', () => {
    expect(
      render({
        deConsumer: true,
        deBrokerName: 'Foo',
        subhead: 'Verify identity',
      })
    ).toMatchSnapshot();
  });

  it('renders logged-in header with firstName', () => {
    expect(render({ loggedIn: true, firstName: 'John' })).toMatchSnapshot();
  });

  it('renders logged-in header without firstName', () => {
    expect(render({ loggedIn: true })).toMatchSnapshot();
  });

  it('renders Spanish header', () => {
    expect(render({ initialLanguage: 'es' })).toMatchSnapshot();
  });

  it('toggles openMenu state when handleMenuToggleClick is called', () => {
    const wrapper = render();

    expect(wrapper.state().openMenu).toBe(false);

    wrapper.instance().handleMenuToggleClick();

    expect(wrapper.state().openMenu).toBe(true);
  });

  it('passes correct props to SkipNav', () => {
    const props = {
      skipNavHref: 'javascript:void(0)',
      onSkipNavClick: () => {},
    };
    const wrapper = render(props);
    const skipNav = wrapper.find(SkipNav);
    expect(skipNav.length).toEqual(1);
    expect(skipNav.props()).toMatchObject({
      href: props.skipNavHref,
      onClick: props.onSkipNavClick,
    });
  });

  it('re-renders with updated links', () => {
    const props = { loggedIn: true };
    const wrapper = render(props);

    let menu = wrapper.find('Menu');
    expect(menu.prop('links').length).toBe(3);

    wrapper.setProps({ links: [{ href: '/foo', label: 'Foo' }] });
    menu = wrapper.find('Menu');

    expect(menu.prop('links').length).toBe(2);
  });

  it('should add spanish toggle if logged in', () => {
    const props = { loggedIn: true };
    const wrapper = render(props);

    const menu = wrapper.find('Menu');
    expect(menu.prop('links').length).toBe(3);

    expect(menu.prop('links')[2].label).toEqual('header.español');
  });

  it('should not add Spanish toggle when logged out', () => {
    const props = {};
    const wrapper = render(props);

    const menu = wrapper.find('Menu');
    expect(menu.prop('links').length).toBe(1);

    expect(menu.prop('links')[0].label).not.toEqual('header.español');
  });

  it('should have "logout" as last item when logged in', () => {
    const props = { loggedIn: true };
    const wrapper = render(props);

    const menu = wrapper.find('Menu');
    const lastLink = menu.dive().find('a').last();

    expect(lastLink).toBeDefined();
    expect(lastLink.text()).toEqual('header.logout');
  });

  it('renders links with absolute URLs if provided a primaryDomain prop', () => {
    expect(render({ primaryDomain: 'https://www.healthcare.gov' })).toMatchSnapshot();
  });
});
