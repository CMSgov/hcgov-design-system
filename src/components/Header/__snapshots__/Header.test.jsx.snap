// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Direct Enrollment banner 1`] = `
<header
  className="hc-c-header hc-c-header--minimal hc-c-header--inverse"
  role="banner"
>
  <SkipNav
    href="#main"
  >
    header.skipNav
  </SkipNav>
  <div
    className="ds-l-container"
  >
    <div
      className="ds-l-row ds-u-align-items--center ds-u-flex-wrap--nowrap ds-u-padding-y--2"
    >
      <a
        className="hc-c-logo-link ds-l-col ds-l-col--auto"
        href="/"
      >
        <Logo
          locale="en"
        />
      </a>
      <div
        className="hc-c-header__subhead hc-c-header__subhead--inline ds-u-md-display--block ds-u-display--none"
      >
        Verify identity
      </div>
      <Component
        deConsumer={true}
        inversed={true}
        locale="en"
        onMenuToggleClick={[Function]}
        open={false}
      />
    </div>
  </div>
  <div
    className="hc-c-header__subhead ds-l-container ds-u-display--block ds-u-md-display--none ds-u-padding-y--1 ds-u-border-top--1 ds-u-border-color--inverse"
  >
    Verify identity
  </div>
  <Menu
    links={
      Array [
        Object {
          "href": "/login?check_de=1",
          "label": "header.login",
        },
        Object {
          "href": "https://www.cuidadodesalud.gov/es/",
          "label": "header.espa単ol",
        },
      ]
    }
    open={false}
  />
  <Component
    deBrokerName="Foo"
  />
</header>
`;

exports[`Header renders Spanish header 1`] = `
<header
  className="hc-c-header hc-c-header--home"
  role="banner"
>
  <SkipNav
    href="#main"
  >
    header.skipNav
  </SkipNav>
  <div
    className="ds-l-container"
  >
    <div
      className="ds-l-row ds-u-align-items--center ds-u-flex-wrap--nowrap ds-u-padding-y--2"
    >
      <a
        className="hc-c-logo-link ds-l-col ds-l-col--auto"
        href="/"
      >
        <Logo
          locale="es"
        />
      </a>
      <Component
        inversed={false}
        locale="es"
        onMenuToggleClick={[Function]}
        open={false}
      />
    </div>
  </div>
  <Menu
    links={
      Array [
        Object {
          "href": "#TODO",
          "label": "TODO: defaultMenuLinks.js",
        },
      ]
    }
    open={false}
  />
</header>
`;

exports[`Header renders full/homepage header 1`] = `
<header
  className="hc-c-header hc-c-header--home"
  role="banner"
>
  <SkipNav
    href="#main"
  >
    header.skipNav
  </SkipNav>
  <div
    className="ds-l-container"
  >
    <div
      className="ds-l-row ds-u-align-items--center ds-u-flex-wrap--nowrap ds-u-padding-y--2"
    >
      <a
        className="hc-c-logo-link ds-l-col ds-l-col--auto"
        href="/"
      >
        <Logo
          locale="en"
        />
      </a>
      <Component
        inversed={false}
        locale="en"
        onMenuToggleClick={[Function]}
        open={false}
      />
    </div>
  </div>
  <Menu
    links={
      Array [
        Object {
          "href": "#TODO",
          "label": "TODO: defaultMenuLinks.js",
        },
      ]
    }
    open={false}
  />
</header>
`;

exports[`Header renders links with absolute URLs if provided a primaryDomain prop 1`] = `
<header
  className="hc-c-header hc-c-header--home"
  role="banner"
>
  <SkipNav
    href="#main"
  >
    header.skipNav
  </SkipNav>
  <div
    className="ds-l-container"
  >
    <div
      className="ds-l-row ds-u-align-items--center ds-u-flex-wrap--nowrap ds-u-padding-y--2"
    >
      <a
        className="hc-c-logo-link ds-l-col ds-l-col--auto"
        href="https://www.healthcare.gov"
      >
        <Logo
          locale="en"
        />
      </a>
      <Component
        inversed={false}
        locale="en"
        onMenuToggleClick={[Function]}
        open={false}
        primaryDomain="https://www.healthcare.gov"
      />
    </div>
  </div>
  <Menu
    links={
      Array [
        Object {
          "href": "#TODO",
          "label": "TODO: defaultMenuLinks.js",
        },
      ]
    }
    open={false}
    primaryDomain="https://www.healthcare.gov"
  />
</header>
`;

exports[`Header renders logged-in header with firstName 1`] = `
<header
  className="hc-c-header hc-c-header--logged-in hc-c-header--inverse"
  role="banner"
>
  <SkipNav
    href="#main"
  >
    header.skipNav
  </SkipNav>
  <div
    className="ds-l-container"
  >
    <div
      className="ds-l-row ds-u-align-items--center ds-u-flex-wrap--nowrap ds-u-padding-y--2"
    >
      <a
        className="hc-c-logo-link ds-l-col ds-l-col--auto"
        href="/"
      >
        <Logo
          locale="en"
        />
      </a>
      <Component
        firstName="John"
        inversed={true}
        locale="en"
        loggedIn={true}
        onMenuToggleClick={[Function]}
        open={false}
      />
    </div>
  </div>
  <Menu
    afterLinks={
      <a
        className="hc-c-menu__link ds-u-border-top--1 ds-u-border--dark ds-u-margin-x--1 ds-u-padding-x--0"
        href="https://www.cuidadodesalud.gov/es/"
      >
        header.espa単ol
      </a>
    }
    beforeLinks={
      <div
        className="ds-u-sm-display--none ds-u-border-bottom--1 ds-u-border--dark ds-u-margin-x--1 ds-u-padding-y--1"
      >
        John
      </div>
    }
    links={
      Array [
        Object {
          "href": "/marketplace/auth/global/en_US/myProfile#landingPage",
          "label": "header.myApplicationsAndCoverage | {\\"lng\\":\\"en\\"}",
        },
        Object {
          "href": "/marketplace/auth/global/en_US/myProfile#settings",
          "label": "header.myProfile | {\\"lng\\":\\"en\\"}",
        },
        Object {
          "href": "/logout",
          "label": "header.logout | {\\"lng\\":\\"en\\"}",
        },
      ]
    }
    open={false}
  />
</header>
`;

exports[`Header renders logged-in header without firstName 1`] = `
<header
  className="hc-c-header hc-c-header--logged-in hc-c-header--inverse"
  role="banner"
>
  <SkipNav
    href="#main"
  >
    header.skipNav
  </SkipNav>
  <div
    className="ds-l-container"
  >
    <div
      className="ds-l-row ds-u-align-items--center ds-u-flex-wrap--nowrap ds-u-padding-y--2"
    >
      <a
        className="hc-c-logo-link ds-l-col ds-l-col--auto"
        href="/"
      >
        <Logo
          locale="en"
        />
      </a>
      <Component
        inversed={true}
        locale="en"
        loggedIn={true}
        onMenuToggleClick={[Function]}
        open={false}
      />
    </div>
  </div>
  <Menu
    afterLinks={
      <a
        className="hc-c-menu__link ds-u-border-top--1 ds-u-border--dark ds-u-margin-x--1 ds-u-padding-x--0"
        href="https://www.cuidadodesalud.gov/es/"
      >
        header.espa単ol
      </a>
    }
    links={
      Array [
        Object {
          "href": "/marketplace/auth/global/en_US/myProfile#landingPage",
          "label": "header.myApplicationsAndCoverage | {\\"lng\\":\\"en\\"}",
        },
        Object {
          "href": "/marketplace/auth/global/en_US/myProfile#settings",
          "label": "header.myProfile | {\\"lng\\":\\"en\\"}",
        },
        Object {
          "href": "/logout",
          "label": "header.logout | {\\"lng\\":\\"en\\"}",
        },
      ]
    }
    open={false}
  />
</header>
`;

exports[`Header renders minimal/product header 1`] = `
<header
  className="hc-c-header hc-c-header--minimal hc-c-header--inverse"
  role="banner"
>
  <SkipNav
    href="#main"
  >
    header.skipNav
  </SkipNav>
  <div
    className="ds-l-container"
  >
    <div
      className="ds-l-row ds-u-align-items--center ds-u-flex-wrap--nowrap ds-u-padding-y--2"
    >
      <a
        className="hc-c-logo-link ds-l-col ds-l-col--auto"
        href="/"
      >
        <Logo
          locale="en"
        />
      </a>
      <div
        className="hc-c-header__subhead hc-c-header__subhead--inline ds-u-md-display--block ds-u-display--none"
      >
        Tax tools
      </div>
      <Component
        inversed={true}
        locale="en"
        onMenuToggleClick={[Function]}
        open={false}
      />
    </div>
  </div>
  <div
    className="hc-c-header__subhead ds-l-container ds-u-display--block ds-u-md-display--none ds-u-padding-y--1 ds-u-border-top--1 ds-u-border-color--inverse"
  >
    Tax tools
  </div>
  <Menu
    links={
      Array [
        Object {
          "href": "/login",
          "label": "header.login",
        },
        Object {
          "href": "https://www.cuidadodesalud.gov/es/",
          "label": "header.espa単ol",
        },
      ]
    }
    open={false}
  />
</header>
`;
